api/client/cp.go

```diff
@@ -3,51 +3,94 @@ package client
 import (
 	"fmt"
 	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
 	"strings"
 
+	"github.com/Sirupsen/logrus"
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/pkg/archive"
 	flag "github.com/docker/docker/pkg/mflag"
 )
 
+type copyFunc func(local, remote, container string, pause bool) error
+
 // CmdCp copies files/folders from a path on the container to a directory on the host running the command.
 //
-// If HOSTDIR is '-', the data is written as a tar file to STDOUT.
+// If LOCALPATH is '-' and the second arg, the data is written as a tar file to STDOUT.
+// If LOCALPATH is '-' and the first arg, the data is sent to the server from STDIN.
 //
-// Usage: docker cp CONTAINER:PATH HOSTDIR
+// Usage: docker cp CONTAINER:PATH|LOCALPATH|- CONTAINER:PATH|LOCALPATH|-
 func (cli *DockerCli) CmdCp(args ...string) error {
-	cmd := cli.Subcmd("cp", "CONTAINER:PATH HOSTDIR|-", "Copy files/folders from a PATH on the container to a HOSTDIR on the host\nrunning the command. Use '-' to write the data as a tar file to STDOUT.", true)
+	cmd := cli.Subcmd("cp", "CONTAINER:PATH|LOCALPATH|- CONTAINER:PATH|LOCALPATH|-", "Copy files/folders to or from the container", true)
```

Comment:
Until we support container <-> container stuff it might be better to write this as something like:
```
[CONTAINER:PATH LOCALPATH|-] | [LOCALPATH|- CONTAINER:PATH]
```